lib/mkFakeDerivation.nix

```diff
@@ -30,7 +34,28 @@
       common
       // rec {
         inherit outputName;
-        outPath = builtins.storePath outputs.${outputName};
+        outPath =
+          # We could be
+          # 1. using an entry from the catalog that has a cache hit
+          # 2. using an entry from the catalog that does not have a cache hit and is
+          #   a. built locally
+          #   b. not built locally
+          # 3. an entry from self - not reproducible, so ultimately we'll throw an error
+          # For case 2a, it would be preferable if we could try builtins.storePath, but we can't, so
+          # just build from source
+          if element ? cache
+          then
+            if builtins.any (cacheMetadata: cacheMetadata.state == "hit") element.cache
+            then
+              if builtins ? fetchClosure
+              then
+                builtins.fetchClosure {
+                  fromStore = (builtins.head element.cache).cacheUrl;
+                  fromPath = outputs.${outputName};
```

Comment:
@tomberek just looked at an old PR you have for this and you have
```
// lib.optionalAttrs (element?build.outputs.${outputName}) {
              toPath = element.build.outputs.${outputName};
            }
```
Will build.outputs be CA? If not, do we still want to use it for `toPath`?

Reply:
Either the convention is that yes, build.outputs is always CA (because for builds this is only known at build-time). or we call it build.ca.outputs and mark it in some way.

For now, yes, use build.outputs.

Another complication to keep in mind: impure-derivations.